HollowKnightComponent.cs

```diff
@@ -1065,14 +1617,35 @@ public class HollowKnightComponent {
                     break;
             }
 
-            return shouldSplit;
+            if (shouldSplit)    splitterAction += 1;
+            if (shouldSkip)     splitterAction += 2;
+            if (shouldReset)    splitterAction += 4; // currently unused but here for future use

```

Comment:
using arithmetic like this to determine actions is a bit too confusing/complex for me to be comfortable introducing into the autosplitter